crates/stackable-shared/src/yaml.rs

```diff
@@ -0,0 +1,155 @@
+//! Utility functions for processing data in the YAML file format
+use std::{io::Write, path::Path, str::FromStr};
+
+use semver::Version;
+use snafu::{ResultExt, Snafu};
+
+const STACKABLE_DOCS_HOME_URL_PLACEHOLDER: &str = "DOCS_BASE_URL_PLACEHOLDER";
+const STACKABLE_DOCS_HOME_BASE_URL: &str = "https://docs.stackable.tech/home";
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Represents every error which can be encountered during YAML serialization.
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("failed to serialize YAML"))]
+    SerializeYaml { source: serde_yaml::Error },
+
+    #[snafu(display("failed to write YAML document separator"))]
+    WriteDocumentSeparator { source: std::io::Error },
+
+    #[snafu(display("failed to write YAML to file"))]
+    WriteToFile { source: std::io::Error },
+
+    #[snafu(display("failed to write YAML to stdout"))]
+    WriteToStdout { source: std::io::Error },
+
+    #[snafu(display("failed to parse {input:?} as semantic version"))]
+    ParseSemanticVersion {
+        source: semver::Error,
+        input: String,
+    },
+
+    #[snafu(display("failed to parse bytes as valid UTF-8 string"))]
+    ParseUtf8Bytes { source: std::string::FromUtf8Error },
+}
+
+pub(crate) struct DocUrlReplacer<'a>(&'a str);
+
+impl<'a> DocUrlReplacer<'a> {
+    pub(crate) fn new(operator_version: &'a str) -> Self {
+        Self(operator_version)
+    }
+
+    fn replace(&self, input: &str) -> Result<String> {
+        let docs_version = match self.0 {
+            "0.0.0-dev" => "nightly".to_owned(),
+            ver => {
+                let v = Version::from_str(ver).context(ParseSemanticVersionSnafu { input })?;
+                format!("{major}.{minor}", major = v.major, minor = v.minor)
+            }
+        };
```

Comment:
Why not parse the version in DocUrlReplacer::new() and store the Version in DocUrlReplacer?

Reply:
Oh wow, what an obvious improvement. So thanks for pointing it out :+1:

Reply:
Done in df719d5

Reply:
Thanks!